### PR TITLE
Add string support for intrinsic min

### PR DESCRIPTION
**Pull request by HarshitaKalani**

### CODE REVIEW

**Comment by HarshitaKalani** [comment]
I thought GFortran compares the strings' lengths and gives the minimum according to the lengths. I just checked again and it does it lexicographically. Will correct this soon.

**Review by Thirumalai-Shaktivel — APPROVED** [review]
The logic seems correct. 
LGTM!

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -3154,8 +3167,16 @@ namespace Min {
                 body.push_back(al, STMT(ASR::make_If_t(al, loc, test,
                     if_body.p, if_body.n, nullptr, 0)));
             }
+        } else if (return_type->type == ASR::ttypeType::Character) {
+            for (size_t i = 1; i < args.size(); i++) {
+                test = make_Compare(make_StringCompare_t, args[i], Lt, result);
```
Comment (Thirumalai-Shaktivel) [inline src/libasr/pass/intrinsic_function_registry.h:3172]: Here, shouldn't we create the loop in the ASR?
Consider we have 5 elements for the `min` function. This creates 4 if condition...
@certik is this fine? the loop seems to be unrolled here.

```clojure
(If
    (IntegerCompare
        (Var 3 x1)
        Lt
        (Var 3 _lcompilers_min0_i32)
        (Logical 4)
        ()
    )
    [(=
        (Var 3 _lcompilers_min0_i32)
        (Var 3 x1)
        ()
    )]
    []
)
(If
    (IntegerCompare
        (Var 3 x2)
        Lt
        (Var 3 _lcompilers_min0_i32)
        (Logical 4)
        ()
    )
    [(=
        (Var 3 _lcompilers_min0_i32)
        (Var 3 x2)
        ()
    )]
    []
)
(If
    (IntegerCompare
        (Var 3 x3)
        Lt
        (Var 3 _lcompilers_min0_i32)
        (Logical 4)
        ()
    )
    [(=
        (Var 3 _lcompilers_min0_i32)
        (Var 3 x3)
        ()
    )]
    []
)
(If
    (IntegerCompare
        (Var 3 x4)
        Lt
        (Var 3 _lcompilers_min0_i32)
        (Logical 4)
        ()
    )
    [(=
        (Var 3 _lcompilers_min0_i32)
        (Var 3 x4)
        ()
    )]
    []
)
(If
    (IntegerCompare
        (Var 3 x5)
        Lt
        (Var 3 _lcompilers_min0_i32)
        (Logical 4)
        ()
    )
    [(=
        (Var 3 _lcompilers_min0_i32)
        (Var 3 x5)
        ()
    )]
    []
)
```

**Review by certik — APPROVED** [review]
Conflicts should be resolved. Otherwise I think this is good to merge.

**Pull request merged**